x/wasm/keeper/authz_policy.go

```diff
@@ -7,16 +7,20 @@ import (
 )
 
 type AuthorizationPolicy interface {
-	CanCreateCode(c types.AccessConfig, creator sdk.AccAddress) bool
+	CanCreateCode(c1 types.AccessConfig, creator sdk.AccAddress, c2 types.AccessConfig, c3 types.AccessConfig) bool
 	CanInstantiateContract(c types.AccessConfig, actor sdk.AccAddress) bool
 	CanModifyContract(admin, actor sdk.AccAddress) bool
 	CanModifyCodeAccessConfig(creator, actor sdk.AccAddress, isSubset bool) bool
 }
 
 type DefaultAuthorizationPolicy struct{}
 
-func (p DefaultAuthorizationPolicy) CanCreateCode(config types.AccessConfig, actor sdk.AccAddress) bool {
-	return config.Allowed(actor)
+func (p DefaultAuthorizationPolicy) CanCreateCode(config types.AccessConfig, actor sdk.AccAddress, instantiateAccess types.AccessConfig, defaultAccessConfig types.AccessConfig) bool {
+	isSubset := true
+	if !instantiateAccess.IsSubset(defaultAccessConfig) {
+		isSubset = false
+	}
+	return config.Allowed(actor) && isSubset
```

Comment:
👍 all decision logic in one place and no "if gov then" logic. nice

Reply:
@alpe yeah I've read your suggestion and it makes more sense.
I'm very appreciative of all the support you've given me.